ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -422,8 +434,12 @@ - (void)setVisibleCoordinateBounds:(MGLCoordinateBounds)bounds edgePadding:(UIEd
 - (void)selectAnnotation:(NSString*)selectedId animated:(BOOL)animated;
 {
     RCTMGLAnnotation * annotation = [_annotations objectForKey:selectedId];
-    if (!annotation) { return; }
-    [_map selectAnnotation:annotation animated:animated];
+    if (!annotation) {
+        _selectedAnnotation = nil;
```

Comment:
If `-objectForKey:` returns no annotation, then the selected annotation does't change, so you shouldn't do this.